src/Utilities/Tuple.hpp

```diff
@@ -174,3 +174,105 @@ constexpr inline void tuple_transform(
       tuple, std::forward<N_aryOp>(op),
       std::make_index_sequence<sizeof...(Elements)>{}, args...);
 }
+
+namespace detail {
+/*!
+ * \brief counts the arguments of a function object.
+ */
+template <typename T>
+struct count_arg;
+
+template <typename R, typename... Args>
+struct count_arg<R(Args...)> {
+  static const size_t value = sizeof...(Args);
+};
+}  // namespace detail
+
+template <typename T, typename... Elements, typename... Args,
+          Requires<sizeof...(Args) == detail::count_arg<T>::value> = nullptr>
+constexpr inline void apply_tuple_combinations(std::tuple<Elements...> /*set*/,
+                                               std::function<T> function,
+                                               Args... args) {
+  function(args...);
+}
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Perform a function over combinatoric subsets of a std::tuple
```

Comment:
The documentation isn't showing up anywhere, and (possibly relatedly) there is a doxygen warning that Travis is not correctly complaining about.  (It is in the output, it just reports as passed.)